trade-finance-manager-ui/server/controllers/utilisation-reports/record-corrections/create-record-correction-request/index.ts

```diff
@@ -0,0 +1,50 @@
+import { Response } from 'express';
+import { getFormattedReportPeriodWithLongMonth } from '@ukef/dtfs2-common';
+import { CreateRecordCorrectionRequestViewModel } from '../../../../types/view-models';
+import { asUserSession } from '../../../../helpers/express-session';
+import { CustomExpressRequest } from '../../../../types/custom-express-request';
+import { PRIMARY_NAVIGATION_KEYS } from '../../../../constants';
+
+export type CreateRecordCorrectionRequestRequest = CustomExpressRequest<{
+  params: {
+    reportId: string;
+    feeRecordId: string;
+  };
+}>;
+
+const renderCreateRecordCorrectionRequestPage = (res: Response, context: CreateRecordCorrectionRequestViewModel) =>
+  res.render('utilisation-reports/record-corrections/create-record-correction-request.njk', context);
+
+export const createRecordCorrectionRequest = (req: CreateRecordCorrectionRequestRequest, res: Response) => {
+  try {
+    const { user } = asUserSession(req.session);
+    const { reportId } = req.params;
+
+    const feeRecordDetails = {
+      bank: {
+        name: 'Test Bank',
+      },
+      reportPeriod: {
+        start: { month: 1, year: 2024 },
+        end: { month: 1, year: 2024 },
+      },
+      facilityId: '0012345678',
+      exporter: 'Sample Company Ltd',
+    };
+
+    return renderCreateRecordCorrectionRequestPage(res, {
```

Comment:
This is the pattern we have for all pages using viewModels, it's nice because it is consistent and also because `res.render` doesn't force you to pass in an object of any particular type, by adding this function and then calling it we are adding in some nice type safety on the object of variables we pass to the template.

Reply:
Thanks @RLavender98 nice pattern - happy to keep for reports, would be great to apply in other areas as there are only 8x instances of `Page = (res`